src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildTriggerTest.java

```diff
@@ -118,9 +128,82 @@ public void check_getters() throws Exception {
             "ciBuildPhrases");
 
     for (String property : properties) {
-      System.out.println(property);
       PropertyType propertyType = descriptor.getPropertyType(property);
       assertThat(propertyType, is(notNullValue()));
     }
   }
+
+  @Test
+  public void constructor_sets_required_properties() throws Exception {
+    StashBuildTrigger stashBuildTrigger =
+        new StashBuildTrigger(cron, stashHost, credentialsId, projectCode, repositoryName);
+
+    assertThat(stashBuildTrigger.getCron(), is(cron));
+    assertThat(stashBuildTrigger.getStashHost(), is(stashHost));
+    assertThat(stashBuildTrigger.getCredentialsId(), is(credentialsId));
+    assertThat(stashBuildTrigger.getProjectCode(), is(projectCode));
+    assertThat(stashBuildTrigger.getRepositoryName(), is(repositoryName));
+  }
+
+  @Test
+  public void start_works_with_valid_credentials() throws Exception {
+    StandardUsernamePasswordCredentials credentials =
+        mock(StandardUsernamePasswordCredentials.class);
+
+    when(credentials.getId()).thenReturn(credentialsId);
+    when(credentials.getUsername()).thenReturn("Username");
+    when(credentials.getPassword()).thenReturn(Secret.fromString("Password"));
```

Comment:
why not just `new UsernamePasswordCredentialsImpl(null, credentialsId, "Username", Secret.fromString("Password") )` instead of mocking?

Reply:
I used that first, but I prefer mocking. `StandardUsernamePasswordCredentials` is an interface. Interfaces are natural seams for mocking. Using a concrete class means ignoring that seam and using the implementation that lies behind it. The `Impl` part has a "private" feel even if it's accessible. We only mock 3 methods, it's not too much code and it's quite readable (more readable than the constructor call, in my opinion).

Reply:
OK, I guess it's just personal preference. I'd rather not use mocks unless there's some complex behavior to record. 

One alternative is implementing the interface inline as an anon class.